ledger/participant-integration-api/src/main/scala/platform/index/ParticipantInMemoryState.scala

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.index
+
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.{Metrics, Timed}
+import com.daml.platform.apiserver.ApiServerConfig
+import com.daml.platform.index.ParticipantInMemoryState.toContractStateEvents
+import com.daml.platform.store.appendonlydao.events.{Contract, ContractStateEvent, Key, Party}
+import com.daml.platform.store.backend.ParameterStorageBackend.LedgerEnd
+import com.daml.platform.store.cache.{EventsBuffer, MutableContractStateCaches}
+import com.daml.platform.store.interfaces.TransactionLogUpdate
+import com.daml.platform.store.interfaces.TransactionLogUpdate.LedgerEndMarker
+
+import java.util.concurrent.Executors
+import scala.concurrent.{ExecutionContext, Future}
+
+// TODO LLP consider pulling in the ledger end cache and string interning view
+class ParticipantInMemoryState(
+    val mutableContractStateCaches: MutableContractStateCaches,
+    val transactionsBuffer: EventsBuffer[TransactionLogUpdate],
+    val completionsBuffer: EventsBuffer[TransactionLogUpdate],
+    updateLedgerApiLedgerEnd: LedgerEnd => Unit,
+    metrics: Metrics,
+) {
+  private val logger = ContextualizedLogger.get(getClass)
+  private val buffersUpdaterExecutionContext =
+    ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor())
+
+  def updateBatch(update: TransactionLogUpdate)(implicit
```

Comment:
adding the update logic to the class with values causing the usual oo friction in factories: the pulling up of the jdbcledgerdao and friends is a consequence that the updateLedgerApiLedgerEnd needs to be instantiated before StandAloneIndexService. Separating data from logic here might allow to put them back